spec/models/status_reaction_spec.rb

```diff
@@ -22,16 +22,23 @@
 
   describe 'before validations' do
     let(:custom_emoji) { Fabricate(:custom_emoji) }
+    let(:remote_custom_emoji) { Fabricate(:custom_emoji, domain: 'example.com') }
     let(:disabled_custom_emoji) { Fabricate(:custom_emoji, disabled: true) }
 
-    context 'when given a custom emoji' do
-      it 'sets custom_emoji' do
+    context 'when given a local custom emoji' do
+      it 'sets custom_emoji to local custom emoji' do
         expect(described_class.create(name: custom_emoji.shortcode, account: account, status: status, custom_emoji: custom_emoji).custom_emoji).to eq custom_emoji
       end
     end
 
+    context 'when given a remote custom emoji' do
+      it 'sets custom_emoji to remote custom emoji' do
+        expect(described_class.create(name: remote_custom_emoji.shortcode, account: account, status: status, custom_emoji: remote_custom_emoji).custom_emoji).to eq remote_custom_emoji
+      end
+    end
+
     context 'when not given a custom emoji' do
-      it 'sets custom_emoji' do
+      it 'sets custom_emoji to custom emoji' do
         expect(described_class.create(name: custom_emoji.shortcode, account: account, status: status, custom_emoji: nil).custom_emoji).to eq custom_emoji
       end
```

Comment:
Honestly, I think it should stay nil, as not sending an emoji object is a violation of spec and this is based on announcement reaction, which don't handle remote content at all.